lib/system/sysio.nim

```diff
@@ -270,12 +270,33 @@ const
     # we always use binary here as for Nim the OS line ending
     # should not be translated.
 
+when defined(posix) and not defined(nimscript):
```

Comment:
Tried that, didn't compile because of circular dependencies probably. After all `sysio` is part of `system`.